src/database.h

```diff
@@ -21,15 +21,23 @@ class Database;
 
 class Database : public Napi::ObjectWrap<Database> {
 public:
+#if NAPI_VERSION < 6
     static Napi::FunctionReference constructor;
+#endif
     static Napi::Object Init(Napi::Env env, Napi::Object exports);
 
     static inline bool HasInstance(Napi::Value val) {
         Napi::Env env = val.Env();
         Napi::HandleScope scope(env);
         if (!val.IsObject()) return false;
         Napi::Object obj = val.As<Napi::Object>();
+#if NAPI_VERSION < 6
         return obj.InstanceOf(constructor.Value());
+#else
+        Napi::FunctionReference* constructor =
+            env.GetInstanceData<Napi::FunctionReference>();
+        return obj.InstanceOf(constructor->Value());
```

Comment:
You should check `constructor` isn't null before dereferencing it.

Reply:
It should never be null though. It's set in the `Init`.

Reply:
What if `HasInstance` is called for a `val` which doesn't have instance data in its `env`?

Reply:
Ah, the val's env would be the same env that we add the instance data to. The environment is shared by all the objects.

Reply:
OK, there's no way of passing an object with a different env - that would have to be from a different worker thread which would require serialization.